src/api/checks/isNodeEligible.ts

```diff
@@ -0,0 +1,53 @@
+import { ZodError } from 'zod';
+import {
+  ApiErrorResponse,
+  IsNodeHealthyPayloadType,
+  IsNodeHealthyResponseType
+} from '../../types';
+import { sdkApiError, fetchWithTimeout, getHeaders } from '../../utils';
+
+/**
+ * Check whether the node is eligible.
+ * @returns A Promise that resolves to a boolean stating that the node is healthy or not.
+ * @throws An error that occurred while processing the request.
+ */
+export const isNodeEligible = async (
+  payload: IsNodeHealthyPayloadType
+): Promise<IsNodeHealthyResponseType> => {
+  const apiEndpointParsed = new URL(payload.apiEndpoint).href;
+  const rawResponse = await fetchWithTimeout(
+    `${apiEndpointParsed}eligiblez`,
+    {
+      method: 'GET',
+      headers: getHeaders(payload.apiToken)
+    },
+    payload.timeout
+  );
+
+  // received unexpected error from server
+  if (rawResponse.status > 499) {
+    throw new Error(rawResponse.statusText);
+  }
+
+  // received expected response
+  if (rawResponse.status === 200) {
+    return true;
+  } else if (rawResponse.status === 412) {
+    return false;
```

Comment:
Why does 412 return false as a result while every other 4xx state would return an APIERror exception?

Reply:
Because 412 is the status saying that the node is not allowed for sure and 200 is allowed for sure. All other statuses are errors

![image](https://github.com/user-attachments/assets/a9f7e4c3-fe2e-4a96-bae5-b8971a2b3f27)